src/server.rs

```diff
@@ -84,8 +108,8 @@ macro_rules! bind_inner {
     (tls: $this:ident, $addr:expr) => {{
         let service = into_service!($this.server.filter);
         let (addr, incoming) = addr_incoming!($addr);
-        let tls = $this.tls.build()?;
-        let srv = HyperServer::builder(crate::tls::TlsAcceptor::new(tls, incoming))
+        let (tls, upgrade) = $this.tls.build()?;
```

Comment:
I could wrap `ServerConfig` in our own struct to include `upgrade`, but returning a tuple should be fine?